recipes/flex/all/conanfile.py

```diff
@@ -24,41 +27,39 @@ class FlexConan(ConanFile):
         "fPIC": True,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def requirements(self):
         self.requires("m4/1.4.19")
 
     def build_requirements(self):
         self.build_requires("m4/1.4.19")
-        if hasattr(self, "settings_build") and tools.cross_building(self):
+        if hasattr(self, "settings_build") and cross_building(self):
             self.build_requires(f"{self.name}/{self.version}")
 
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Flex package is not compatible with Windows. "
+                                            "Consider using winflexbison instead.")
+
     def config_options(self):
         if self.settings.os == "Windows":
```

Comment:
looking at the `validate()` block above, I feel there's no scenario in which the logic in the `config_options()` method will be executed, since the windows is altogether unsupported. I'd consider a removal :D